### ISSUE TITLE
Injecting a x86 target with a x86 dll from a x64 injector

### ISSUE THREAD

**Issue by dvv47** [EasyHook/EasyHook#94]
When i try this, EasyHook return me error:
It is not supported to directly hook through the WOW64 barrier.

But as i read, its possible to inject x86 and x64 dll from x64 app.
Here is two links:
http://stackoverflow.com/questions/29652839/injecting-a-x86-target-with-a-x86-dll-from-a-x64-injector
http://stackoverflow.com/questions/13297452/x86-code-injection-into-an-x86-process-from-a-x64-process

**Comment by spazzarama** [comment]
@dvv47 you can inject into 32-bit process from a 64-bit process if using EasyHook in .NET. You also need to have the correct binaries in place to be able to do this (use the NuGet package if in doubt).

Can you provide some more detail about the error you received?

**Comment by dvv47** [comment]
@spazzarama i want to inject native dll to process. And this error i got when trying inject from native application.
i am also tried inject native dll from .NET application, but in this case it throw Wrong binary error

**Comment by spazzarama** [comment]
The process to get around the WOW64 barrier currently only works for .NET, I'll mark this as a feature request.

**Comment by venkatay** [comment]
Hello All,

Wanted to check if this is is fixed. I am using a gaming anywhere application which uses easy hook to inject into a 64bit application. 

Thanks,
Venkat

**Comment by spazzarama** [comment]
@venkatay hooking 32-bit from 64-bit is supported if your app is .NET

**Comment by venkatay** [comment]
Yes the app is .NET. Also i corrected the app to make it 64bit. It launches the exe in the suspended mode but when it calls NtCreateThreadEx(..) it gets killed.

**Comment by spazzarama** [comment]
@venkatay does it work if not using CreateAndInject? Test with some simple targets as well. Just a heads up that from memory if the target app is .NET as well the CreateAndInject won't work, I doubt the problem is not that it is a 32- or 64-bit target, perhaps provide some more details in a new issue.

**Comment by Rockford123** [comment]
Is this feature added in latest release? I am trying to inject x86 dll in x86 application using x64 bit injector.

Thanks.

**Comment by spazzarama** [comment]
If using .NET it already worked, if you are only using the native easyhook library it does not support bypassing the WOW64 boundary.
